contracts/Staking.sol

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "hardhat/console.sol";
+
+contract Staking is ERC721URIStorage, Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
+
+    bool public active;
+    uint256 public startTime;
+    uint256 public endTime;
+    uint256 public initialised;
+    uint256 public TOTAL_SEURO;
+    uint256 public SEURO_REMAINING;
+    uint public SEUROTST;
+    uint public INTEREST;
+
+    uint256 private minTST;
+
+    address TST_ADDRESS;
+    address SEURO_ADDRESS;
+
+    mapping(address => Position) private _positions;
+
+    struct Position {
+        uint96 nonce;
+        uint256 tokenId;
+        bool open;
+        uint256 totalValue;
+        uint256 reward;
+    }
+
+    constructor(string memory _name, string memory _symbol) ERC721(_name, _symbol) {}
+
+    function activate(
+        uint256 _start,
+        uint256 _end,
+        address _TST_ADDRESS,
+        address _SEURO_ADDRESS,
+        uint256 _TOTAL_SEURO,
+        uint _SEUROTST,
+        uint _INTEREST
+    ) external onlyOwner {
+
+        // CHORE needs refactor
+        require(active == false, 'err-already-active');
+        require(initialised == 0, 'err-already-initialised');
+        require(_end > _start, 'err-start-end');
+        require(_end >= block.timestamp, 'err-invalid-end');
+
+        TOTAL_SEURO = _TOTAL_SEURO;
+        SEURO_REMAINING = _TOTAL_SEURO;
+        SEUROTST = _SEUROTST;
+        INTEREST = _INTEREST;
+        TST_ADDRESS = _TST_ADDRESS;
+        SEURO_ADDRESS = _SEURO_ADDRESS;
+        startTime = _start;
+        endTime = _end;
+        initialised = block.timestamp;
+
+        // TODO variable
+        minTST = 1 ether;
+
+        active = true;
+    }
+
+    function disable() external onlyOwner {
+        require(active, 'err-not-active');
+        active = false;
+    }
+
+    // reward works out the amount of seuro given to the user based on the
+    // amount of TST they first put in.
+    function reward(uint256 _amount) public view returns (uint256) {
+        uint256 SEURO = _amount * SEUROTST / 10_000;
+        uint256 REWARD = SEURO * INTEREST / 10_000;
+        return REWARD + SEURO;
+    }
+
+    function mint(uint256 _amount) external returns(uint256) {
```

Comment:
Mint is also part of the 721 spec, so I followed suit. Others, include UNI for example, 'mint' a new position too. 

Since we're minting and burning for a reward, we would need to change burn also.

Reply:
fair enough